### PR TITLE
Fixes  #1317: Reducers and Actions cleaned

### LINKED ISSUE #1317: Cleanup reducers and actions
**I'm submitting a ...**
- [ ] bug report
- [ ] feature request
- [x] code cleanup

**Current behavior:**
The reducers and actions currently contain unused code and old JS standards such as Object.assign

**Expected behavior:**
Remove unused statements, structure the code in a logical fashion and replace Object.assign with spread operator.

**Will work on the issue:**
Yes

### PR DESCRIPTION
**Pull request by AakashMallik**
Fixes #1317 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `master` branch.
- [ ] I have added necessary documentation (if appropriate)
- [x] Added Surge preview link

#### Changes proposed in this pull request:

- Cleaned up unused code. 
- Replaced Object.assign with spread operator
- Removed unnecessary line breaks.
- Restructured reduces/index.js in a logical fashion

- https://pr-1318-fossasia-susper.surge.sh

### CODE REVIEW

File: src/app/reducers/knowledge.ts [hunk 1]
```diff
@@ -1,54 +1,51 @@
...
-export const CONTENT_CHANGE = 'CONTENT_CHANGE';
-export const IMAGE_CHANGE = 'IMAGE_CHANGE';
-
 export interface State {
   content_response: any,
   image_response: any,
   description_response: any
 }
 
-const initialState: State = {
```
Comment (rajvaibhavdubey) [inline src/app/reducers/knowledge.ts]: I don't think that changing the variable name here was required , both `defaultState` and `inititalState` mean the same .

Comment (AakashMallik) [reply src/app/reducers/knowledge.ts]: Actually, it happens to be the convention in redux to I applied it here too. @rajvaibhavdubey  Do the other changes look okay?

**Comment by AakashMallik** [comment]
@praveenojha33 I was running the same command that travis ran on my machine but it worked perfectly there, and but travis is throwing error on that same command. I'll have to dig a bit deeper to find out.

File: src/app/reducers/index.ts [hunk 2]
```diff
@@ -85,9 +87,14 @@ export const getSpeechState = (state: State) => state.speech;
 
 export const getSearchResults = createSelector(getSearchState, fromSearch.getsearchresults);
 export const getItems = createSelector(getSearchState, fromSearch.getItems);
+export const getResponseTime = createSelector(getSearchState, fromSearch.getresponsetime);
 export const getTotalResults = createSelector(getSearchState, fromSearch.getTotalResults);
 export const getNavigation = createSelector(getSearchState, fromSearch.getNavigation);
+export const getSearchStatus = createSelector(getSearchState, fromSearch.getSearchstatus);
+
```
Comment (samagragupta) [inline src/app/reducers/index.ts]: Why there is an extra line?

Comment (AakashMallik) [reply src/app/reducers/index.ts]: @samagragupta Probably because of tslint.

**Comment by shreyanshdwivedi** [comment]
@AakashMallik I'll suggest you to open and close this PR. This is a glitch
Idk exactly why this happens so frequently but it happens to me too. If you find out real problem and solve it, it'll be great. Otherwise you can fix travis failure by trying to run travis again

**Comment by Pipe-Runner** [comment]
@shreyanshdwivedi  Thanks pal, it actually worked. @praveenojha33 Let me know if this is fine by you.

**Pull request merged**